src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityPassthroughHatchFluid.java

```diff
@@ -101,40 +132,68 @@ protected IItemHandlerModifiable createImportItemHandler() {
-                                .setContainerClicking(true, true)
-                                .setAlwaysShowFull(true));
+
+        int backgroundWidth = 9 * 18 + 14;
+        int backgroundHeight = 18 + 18 * rowSize + 94;
+
+        List<List<IWidget>> widgets = new ArrayList<>();
+        for (int i = 0; i < rowSize; i++) {
+            widgets.add(new ArrayList<>());
+            for (int j = 0; j < rowSize; j++) {
+                widgets.get(i).add(new GTFluidSlot().syncHandler(fluidTankList.getTankAt(i * rowSize + j))
+                        .background(GTGuiTextures.FLUID_SLOT));
             }
         }
-        return builder.bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7, 18 + 18 * rowSize + 12);
+
+        BooleanSyncValue workingStateValue = new BooleanSyncValue(() -> workingEnabled, val -> workingEnabled = val);
+        guiSyncManager.syncValue("working_state", workingStateValue);
+
+        // TODO: Change the position of the name when it's standardized.
+        return GTGuis.createPanel(this, backgroundWidth, backgroundHeight)
+                .child(IKey.lang(getMetaFullName()).asWidget().pos(5, 5))
+                .child(SlotGroupWidget.playerInventory().left(7).bottom(7))
+                .child(new Grid()
+                        .top(18).height(rowSize * 18)
+                        .minElementMargin(0, 0)
+                        .minColWidth(18).minRowHeight(18)
+                        .alignX(0.5f)
+                        .matrix(widgets))
+                .child(new ToggleButton()
+                        .top(18 * 2).left(18 * 8 + 7)
+                        .value(new BoolValue.Dynamic(workingStateValue::getBoolValue,
+                                workingStateValue::setBoolValue))
+                        .overlay(GTGuiTextures.BUTTON_FLUID_OUTPUT)
+                        .tooltipBuilder(t -> t.setAutoUpdate(true)
+                                .addLine(workingStateValue.getBoolValue() ?
+                                        IKey.lang("gregtech.gui.fluid_passthrough.enabled") :
+                                        IKey.lang("gregtech.gui.fluid_passthrough.disabled"))));
     }
 
     @Override
     public NBTTagCompound writeToNBT(NBTTagCompound tag) {
         super.writeToNBT(tag);
         tag.setTag("FluidInventory", fluidTankList.serializeNBT());
+        tag.setBoolean("WorkingEnabled", workingEnabled);
         return tag;
     }
 
     @Override
     public void readFromNBT(NBTTagCompound tag) {
         super.readFromNBT(tag);
         this.fluidTankList.deserializeNBT(tag.getCompoundTag("FluidInventory"));
+        // Passthrough hatches before this change won't have WorkingEnabled at all, so we need to check if it exists
+        if (tag.hasKey("WorkingEnabled")) {
+            this.workingEnabled = tag.getBoolean("WorkingEnabled");
+        } else {
+            this.workingEnabled = true;
+        }
```

Comment:
this else isn't really necessary, as it is set to true already in the contructor